src/alg/eddsa_dalek.rs

```diff
@@ -71,7 +70,7 @@ impl VerifyingKey<Ed25519> for ed25519_dalek::VerifyingKey {
     fn from_slice(raw: &[u8]) -> anyhow::Result<Self> {
         let raw: &[u8; PUBLIC_KEY_LENGTH] = raw
             .try_into()
-            .context("Ed25519 public key has unexpected length")?;
+            .map_err(|err: core::array::TryFromSliceError| anyhow::anyhow!(err))?;
```

Comment:
Bikeshedding: I'd prefer to provide a more specific error message (like the one used with `context()`) so that they are more easily debuggable. Also, the error type annotation can be avoided if `try_into` is replaced with `try_from`:

```rust
let raw = <&[u8; PUBLIC_KEY_LENGTH]>::try_from(raw)
    .map_err(|err| anyhow::anyhow!(err))
    .context("Ed25519 public key has unexpected length")?;
```